content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md

```diff
@@ -0,0 +1,242 @@
+
+h﻿ere is a small extract of the `install_backend.yml` playbook: Full version [here](https://github.com/Workshops-on-Demand/wod-backend/blob/main/ansible/install_backend.yml)
+
+```shellsession
+vi install_backend
+- hosts: all
+  gather_facts: true
+  vars:
+    IJAVAVER: "1.3.0"
+    KUBECTLVER: "1.21.6"
+
+  tasks:
+    - name: Include variables for the underlying distribution
+      include_vars: "{{ ANSIBLEDIR }}/group_vars/{{ ansible_distribution }}-{{ ansible_distribution_major_version }}.yml"
+
+    - name: Base setup for a JupyterHub environment server or appliance
+      include_tasks: "{{ ANSIBLEDIR }}/setup_base_appliance.yml"
+
+    - name: Add CentOS SC repository into repo list
+      become: yes
+      become_user: root
+      yum:
+        name: centos-release-scl-rh
+        state: present
+      when:
+        - ansible_distribution == "CentOS"
+        - ansible_distribution_major_version >= "7"
+
+    - name: Add conda GPG Key to APT
+      become: yes
+      become_user: root
+      apt_key:
+        url: https://repo.anaconda.com/pkgs/misc/gpgkeys/anaconda.asc
+        state: present
+      when:
+       - ansible_distribution == "Ubuntu"
+       - ansible_distribution_major_version >= "20"
+
+      # TODO: Do it for EPEL if really needed
+    - name: Add conda APT repository
+      become: yes
+      become_user: root
+      apt_repository:
+        repo: deb [arch=amd64] https://repo.anaconda.com/pkgs/misc/debrepo/conda stable main
+        state: present
+      when:
+       - ansible_distribution == "Ubuntu"
+```
+
+P﻿ossible Use Cases:
+
+* U﻿pgrade to a newer version of Jupyterhub
+* A﻿dd a new kernel to Jupyterhub
+* A﻿dd a new Ansible Galaxy collection
+* A﻿dd a new PowerShell library 
+* A﻿dd a new package needed by a workshop. 
+
+For e.g:
+
+* Kubectl client  
+* T﻿erraform client
+* P﻿owerShell module  
+* P﻿ython Library
+
+You will start by moving to your public backend forked repository and apply the necessary changes before committing and push locally. 
+
+Then you will perform a merge request with the main repository. We plan to integrate here  in a proper CICD (continuous integration continous development) pipeline to allow a vagrant based test deployment. Whenever someone performs a merge request on the main repo, the test deployment task kicks in and deploy a virtual backend server on which the new version of the installation process is automatically tested. When successful, the merge request is accepted. Once merged, you will need to move to your backend server and perform git remote update and git rebase on the wod-backend directory. Once done, you will then be able to perform the installation process.
```

Comment:
In the middle of the 3rd sentence, it should read "the test deployment task kicks in and deploys..." (s at the end of deploy)